asyncpg/types.py

```diff
@@ -63,10 +63,10 @@ class _RangeValue(typing.Protocol):
     def __eq__(self, __value: object) -> bool:
         ...
 
-    def __lt__(self, __other: _RangeValue) -> bool:
+    def __lt__(self: _RV, __other: _RV) -> bool:
```

Comment:
I believe `def __lt__(self, __other: Self) -> bool:` should work

Reply:
```suggestion
    def __lt__(self, __other: Self, /) -> bool:
```

This will fix it, checked locally with `pyright` as well.

See `typeshed` which has the definition of `int.__lt__` that `pyright` (correctly) consider to not be compatible with `_RangeValue.__lt__`:
https://github.com/python/typeshed/blob/ca65e087f1f9dfc28e89192b60ce7cfc2e92c674/stdlib/builtins.pyi#L321